src/main/java/com/iexec/core/replicate/ReplicatesService.java

```diff
@@ -245,40 +241,32 @@ public void updateReplicateStatus(String chainTaskId,
         updateReplicateStatus(chainTaskId, walletAddress, statusUpdate);
     }
 
-    /*
-     * We retry up to 100 times in case the task has been modified between
-     * reading and writing it.
+    /**
+     * Checks whether a replicate can be updated with given status.
      *
-     * Before updating we check:
-     *   1) if valid transition.
-     *   2) if worker did fail when CONTRIBUTE/REVEAL/UPLOAD_FAILED.
-     *   3) if worker did succeed onChain when CONTRIBUTED/REVEALED.
-     *   4) if worker did upload when RESULT_UPLOADING.
+     * @return {@link Optional#empty()} if this update is OK, {@code Optional} containing the error reason otherwise.
      */
-    @Retryable(value = {OptimisticLockingFailureException.class}, maxAttempts = 100)
-    public Optional<TaskNotificationType> updateReplicateStatus(String chainTaskId,
-                                                                String walletAddress,
-                                                                ReplicateStatusUpdate statusUpdate) {
-        log.info("Replicate update request [status:{}, chainTaskId:{}, walletAddress:{}, details:{}]",
-                statusUpdate.getStatus(), chainTaskId, walletAddress, statusUpdate.getDetailsWithoutStdout());
-
+    public Optional<ReplicateStatusUpdateError> canUpdateReplicateStatus(String chainTaskId,
+                                               String walletAddress,
+                                               ReplicateStatusUpdate statusUpdate) {
         Optional<ReplicatesList> oReplicateList = getReplicatesList(chainTaskId);
         if (oReplicateList.isEmpty() || oReplicateList.get().getReplicateOfWorker(walletAddress).isEmpty()) {
-            log.error("Cannot update replicate, could not get replicate [chainTaskId:{}, UpdateRequest:{}]",
-                    chainTaskId, statusUpdate);
-            return Optional.empty();
+            return Optional.of(ReplicateStatusUpdateError.UNKNOWN_REPLICATE);
         }
 
         ReplicatesList replicatesList = oReplicateList.get();
-        Replicate replicate = replicatesList.getReplicateOfWorker(walletAddress).get();
+        Replicate replicate = replicatesList.getReplicateOfWorker(walletAddress).orElseThrow();
```

Comment:
This is already checked above in
```
... || oReplicateList.get().getReplicateOfWorker(walletAddress).isEmpty()
```

Reply:
Sure, but using only `get` triggers a warning in IntelliJ.
By the way, `geŧ` and `orElseThrow` have the same implementation - at least in OpenJDK 11:
```
    public T get() {
        if (value == null) {
            throw new NoSuchElementException("No value present");
        }
        return value;
    }
```
So I prefer to use `orElseThrow` there.

Reply:
I think it is more a matter of readability. When `orElseThrow()` is used, that implicitly hints about the possibility of the value being empty which is impossible in this case. `get()` however implies that the reader can continue his way peacefully, the value will be there.

Reply:
I didn't know about IntelliJ warnings though.

Reply:
IntelliJ warnings can be ignored with a comment (`//noinspection OptionalGetWithoutIsPresent`) but I don't think it is good solution when using different IDEs.
By the way, it also triggers a major warning in SonarLint:
![image](https://user-images.githubusercontent.com/87655102/130631129-fbdea0f6-9cdc-49b2-96ec-3dda68d8c9f5.png)

Would you mind if we keep the `orElseThrow` call with a comment such as `//  "get" could be used there but triggers a warning`?

Reply:
Yes, no problem :+1: . Thank you for your patience.